File: 3895-count-digit-appearances/3895-count-digit-appearances.go
func countDigitOccurrences(nums []int, digit int) int {
    count := 0
    for _, num:= range nums{
        var n = num
        for n>0 {
            d := n%10
            if d==digit{
                count++
            }
            n/=10
        } 
    }
    return count
}

File: 3895-count-digit-appearances/3895-count-digit-appearances.java
class Solution {
    public int countDigitOccurrences(int[] nums, int digit) {
        int count = 0;
        for(int num: nums){
            int n1 = num;
            while(n1>0){
                int d = n1%10;
                if(d == digit){
                    count++;
                }
                n1/=10;
            }
        }
        return count;
    }
}